.github/workflows/ci-tests.yml

```diff
@@ -1,31 +1,73 @@
 name: "ci tests"
-on: [push, pull_request]
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - 'README.md'
+      - 'ARCHITECTURE.md'
```

Comment:
Never used these, but I can see them as beneficial in other repos as well :+1:

Reply:
Yes, the idea came to me when I remembered my experience on updating the documentation in subspace/subspace repo, and it was taking hours to pass the test, although not a single code line is changed...